component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/OneTimePassword.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.authenticator.voiceotp;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * This class contains util methods to generate an OTP.
+ *
+ * Important: These methods are duplicated in the common module's
+ * 'org.wso2.carbon.identity.voiceotp.common.util.OneTimePasswordUtils' class as well.
+ * Any fixes here, should be reflected there as well.
+ */
+public class OneTimePassword {
```

Comment:
If this class contains the Util methods, Shouldn't the classname has to be **OneTimePasswordUtil** ?

Reply:
This is the same implementation & name which having in the smsotp connector. If required, we can change this to OneTimePasswordUtil.

[1]https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/master/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/OneTimePassword.java

Reply:
I would be more appropriate to use **OneTimePasswordUtil**. So let's change it.